app/src/main/java/com/example/concordia_campus_guide/Fragments/LocationFragment/LocationFragment.java

```diff
@@ -330,11 +332,15 @@ public boolean setupBuildingMarkerClickListener(GoogleMap map) {
         map.setOnMarkerClickListener(new GoogleMap.OnMarkerClickListener() {
             @Override
             public boolean onMarkerClick(Marker marker) {
-                Building building = mViewModel.getBuildingFromeCode(marker.getTag().toString());
-                //TODO: Make function that pops up the info card for the building (via the building-code)
-                String buildingCode = (marker.getTag()).toString();
-                ((MainActivity) getActivity()).showInfoCard(buildingCode);
-                onBuildingClick(building);
+                if (marker.getTag().toString().contains(POI_TAG)) {
+                    //TODO: Use the directions api to get directions to this point of interest
+                    marker.showInfoWindow();
```

Comment:
What does this function do?

Reply:
When you click on a marker, if it's a POI marker, then it should show the directions to that POI.